### PR TITLE
OOP1 Homework1 (SHM1)

### PR DESCRIPTION
**Pull request by PiotrHCpp**

### CODE REVIEW

File: shm/Ship.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        : Ship(0, maxCrew, speed, "", id)
+    {}
+
+    Ship& Ship::operator-=(size_t num) {
+        crew_ -= num;
+        return *this;
+    }
+    Ship& Ship::operator+=(size_t num) {
+        crew_ += num;
+        return *this;
```
Comment (WBigaj) [inline shm/Ship.cpp:26]: What if you would try to add crewman beyond maxCrew size?

File: shm/Ship.cpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        , id_(id)
+    {}
+
+    Ship::Ship(int maxCrew, int speed, size_t id)
+        : Ship(0, maxCrew, speed, "", id)
+    {}
+
+    Ship& Ship::operator-=(size_t num) {
+        crew_ -= num;
+        return *this;
```
Comment (WBigaj) [inline shm/Ship.cpp:22]: What if you would try to remove crewman bellow zero?

**Review by WBigaj — COMMENTED** [review]
I have checked your code and given my feedback

File: shm/Cargo.cpp [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    : name_(name), amount_(amount), basePrice_(basePrice)
+{}
+
+Cargo& Cargo::operator+=(const size_t& amount) {
+    amount_ += amount;
+    return *this;
+}
+Cargo& Cargo::operator-=(const size_t& amount) {
+    amount_ -= amount;
+    return *this;
```
Comment (nauka-programowania-MA) [inline shm/Cargo.cpp:13]: what if amount drop below 0?

File: shm/Cargo.cpp [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+}
+bool Cargo::operator==(const Cargo& cargo) const {
+    return (name_ == cargo.name_);
```
Comment (nauka-programowania-MA) [inline shm/Cargo.cpp:16]: () redundant

File: shm/Coordinates.cpp [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    : positionX_(0)
+    , positionY_(0)
+{}
+Coordinates::Coordinates(int positionX, int positionY)
+    : positionX_(positionX)
+    , positionY_(positionY)
+{}
+
+bool Coordinates::operator==(const Coordinates& coords) {
+    return ((positionX_ == coords.positionX_) && (positionY_ == coords.positionY_));
```
Comment (nauka-programowania-MA) [inline shm/Coordinates.cpp:13]: () redundant

File: shm/Island.cpp [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+    Island::Island()
+        : position_(Coordinates(0,0))
+    {}
+    Island::Island(int positionX, int positionY)
+        : position_(Coordinates(positionX, positionY))
+    {}
+
+    bool Island::operator==(const Island& island) {
+        return (*this).getPosition() == island.getPosition();
```
Comment (nauka-programowania-MA) [inline shm/Island.cpp:11]: (*this) redundant, just call `getPosition()`

File: shm/Map.cpp [hunk 7]
```diff
@@ -0,0 +1,37 @@
+#include "Map.hpp"
+
+#include <algorithm>
+#include <random>
+
+Map::Map() : sizeX_(10), sizeY_(10) {
+    islandGenerator(10);
```
Comment (nauka-programowania-MA) [inline shm/Map.cpp:7]: Maybe as constexpr value. To get know what this numbers means.

File: shm/Map.hpp [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    Map(size_t sizeX, size_t sizeY, size_t islandAmount);
+
+private:
+    void islandGenerator(size_t amount);
+    Island* getIsland(const Coordinates& coordinates);
+
+    size_t sizeX_;
+    size_t sizeY_;
+    std::vector<Island> islandsArrangement_;
+    std::shared_ptr<Island> currentPosition_{};
```
Comment (nauka-programowania-MA) [inline shm/Map.hpp:20]: You don't need to initialize it here.

File: shm/Ship.cpp (hunk 2 above)
Comment (nauka-programowania-MA) [reply shm/Ship.cpp:22]: +1

File: shm/Ship.cpp (hunk 1 above)
Comment (nauka-programowania-MA) [reply shm/Ship.cpp:26]: +1

File: shm/Ship.cpp [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    size_t Ship::getId() const        { return id_; }
+    size_t Ship::getMaxCrew() const   { return maxCrew_; }
+    std::string Ship::getName() const { return name_; }
+    int Ship::getSpeed() const        { return speed_; }
+    Cargo* Ship::getCargo(size_t index) const {
+        return cargo_[index].get();
+    }
+    std::vector<std::shared_ptr<Cargo>> Ship::getCargos() const {
+        return cargo_;
+    }
```
Comment (nauka-programowania-MA) [inline shm/Ship.cpp:40]: One style

**Review by nauka-programowania-MA — APPROVED** [review]

**Comment by nauka-programowania-MA** [comment]
{
"requirement_ids": [2568, 2568, 2568, 2568, 2568, 2568, 2568, 2568],
"points": [5, 5, 5, 5, 5, 5, 5, 5]
}

**Pull request closed without merging**